docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+    identifiers defined in this specification. If an implementation
+    does not recognise the extension, but a ``fallback`` is present,
+    then the implementation may proceed using the ``fallback`` value
+    as the data type. For fallback types that do not correspond to base
+    known types, extensions can fallback on a raw number of bytes using
+    the raw type (``r*``).
+
+``chunk_grid``
+^^^^^^^^^^^^^^
+
+    The chunk grid of the Zarr array. If the chunk grid is a regular
+    chunk grid as defined in this specification, then the value must
+    be an object with the names ``type`` and ``chunk_shape``. The
+    value of ``type`` must be the string ``"regular"``, and the value of
+    ``chunk_shape`` must be an array of integers providing the lengths
+    of the chunk along each dimension of the array. For example,
+    ``{"type": "regular", "chunk_shape": [2, 5], "separator":"/"}`` means a regular
+    grid where the chunks have length 2 along the first dimension and
+    length 5 along the second dimension.
+
+    The ``chunk_grid`` value is an extension point and may be defined
+    by an extension. If the chunk grid type is defined by an
+    extension, then the value must be an object containing
+    the names ``extension`` and ``type``. The ``extension`` is
+    required and the value must be a URI that identifies the
+    extension and dereferences to a human-readable representation of
+    the specification.  The ``type`` is required and the value is
+    defined by the extension.
+
+``chunk_memory_layout``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+    The internal memory layout of the chunks. Use the value "C" to
+    indicate `C contiguous memory layout`_ or "F" to indicate
+    `F contiguous memory layout`_ as defined in this specification.
+
+    The ``chunk_memory_layout`` value is an extension point and may be
+    defined by an extension. If the chunk memory layout type
+    is defined by an extension, then the value must be an
+    object containing the names ``extension`` and ``type``. The
+    ``extension`` is required and the value must be a URI that
+    identifies the extension and dereferences to a
+    human-readable representation of the specification.  The ``type`` is
+    required and the value is defined by the extension.
+
+``fill_value``
+^^^^^^^^^^^^^^
+
+    Provides an element value to use for uninitialised portions of the
+    Zarr array.
+
+    If the data type of the Zarr array is Boolean then the value must
+    be the literal ``false`` or ``true``. If the data type is one of
+    the integer data types defined in this specification, then the
+    value must be a number with no fraction or exponent part and must
+    be within the range of the data type.
+
+    For any data type, if the ``fill_value`` is the literal ``null``
+    then the fill value is undefined and the implementation may use
+    any arbitrary value that is consistent with the data type as the
+    fill value.
```

Comment:
Using `fill_value: null` to represent "there is no fill value" (as opposed to "the fill value is `null`") can cause confusion when the data type of an array includes `None` / `null` as a value. See https://github.com/zarr-developers/zarr-specs/pull/145. This could be addressed either by making `fill_value` mandatory (my preference), or representing "is there a fill value" separately from "the fill value is {value}" in metadata.

Reply:
This is already addressed in:
https://github.com/zarr-developers/zarr-specs/pull/145/

I think that should just be merged into here.

Reply:
Marking as resolved, since #145 is merged into main, which represents the current status